src/eval.rs

```diff
@@ -204,14 +204,26 @@ pub fn eval_main<'tcx>(tcx: TyCtxt<'tcx>, main_id: DefId, config: MiriConfig) ->
     // Perform the main execution.
     let res: InterpResult<'_, i64> = (|| {
         // Main loop.
-        while ecx.step()? {
+        loop {
+            match ecx.schedule()? {
+                SchedulingAction::ExecuteStep => {
+                    assert!(ecx.step()?, "a terminated thread was scheduled for execution");
+                }
+                SchedulingAction::ExecuteDtors => {
+                    ecx.run_tls_dtors_for_active_thread()?;
+                }
+                SchedulingAction::Stop => {
+                    break;
+                }
+            }
             ecx.process_diagnostics();
         }
         // Read the return code pointer *before* we run TLS destructors, to assert
         // that it was written to by the time that `start` lang item returned.
         let return_code = ecx.read_scalar(ret_place.into())?.not_undef()?.to_machine_isize(&ecx)?;
-        // Global destructors.
-        ecx.run_tls_dtors()?;
+        // Run Windows destructors. (We do not support concurrency on Windows
+        // yet, so we run the destructor of the main thread separately.)
+        ecx.run_windows_tls_dtors()?;
```

Comment:
Even if Windows always has just one thread, isn't there a good way to unify the destructors?

I feel like running on Windows should be equivalent to running on macOS/Linux and just never spawning a second thread.